sim/rogue/item_sets_phase_8.go

```diff
@@ -13,3 +18,126 @@ func (rogue *Rogue) ApplyFallenRegalityRogueBonus(aura *core.Aura) {
 		})
 	})
 }
+
+var ItemSetDuskwraithLeathers = core.NewItemSet(core.ItemSet{
+	Name: "Duskwraith Leathers",
+	Bonuses: map[int32]core.ApplyEffect{
+		2: func(agent core.Agent) {
+			rogue := agent.(RogueAgent).GetRogue()
+			rogue.applyScarletEnclaveTank2PBonus()
+		},
+		4: func(agent core.Agent) {
+			rogue := agent.(RogueAgent).GetRogue()
+			rogue.applyScarletEnclaveTank4PBonus()
+		},
+		6: func(agent core.Agent) {
+			rogue := agent.(RogueAgent).GetRogue()
+			rogue.applyScarletEnclaveTank6PBonus()
+		},
+	},
+})
+
+// Your stacks of Rolling with the Punches also increase all damage you deal by 1%.
+func (rogue *Rogue) applyScarletEnclaveTank2PBonus() {
+
+	if !rogue.HasRune(proto.RogueRune_RuneRollingWithThePunches) {
+		return
+	}
+
+	label := "S03 - Item - Scarlet Enclave - Rogue - Tank 2P Bonus"
+	if rogue.HasAura(label) {
+		return
+	}
+
+	rogue.RegisterAura(core.Aura{
+		Label: label,
+		OnInit: func(aura *core.Aura, sim *core.Simulation) {
+			oldOnStacksChange := rogue.RollingWithThePunchesProcAura.OnStacksChange
+			rogue.RollingWithThePunchesProcAura.OnStacksChange = func(aura *core.Aura, sim *core.Simulation, oldStacks int32, newStacks int32) {
+				oldOnStacksChange(aura, sim, oldStacks, newStacks)
+				rogue.PseudoStats.DamageDealtMultiplier /= 1 + 0.01*float64(oldStacks)
+				rogue.PseudoStats.DamageDealtMultiplier *= 1 + 0.01*float64(newStacks)
+			}
```

Comment:
🔨 Small knit here. We have a helper that simplifies this `Aura#ApplyOnStacksChange`

```suggestion
			rogue.RollingWithThePunchesProcAura.ApplyOnStacksChange(func(aura *core.Aura, sim *core.Simulation, oldStacks int32, newStacks int32) {
				oldOnStacksChange(aura, sim, oldStacks, newStacks)
				rogue.PseudoStats.DamageDealtMultiplier /= 1 + 0.01*float64(oldStacks)
				rogue.PseudoStats.DamageDealtMultiplier *= 1 + 0.01*float64(newStacks)
			})
```

Reply:
Just to make sure, I don't need to call `oldOnStacksChange` with this helper right? It chains whatever callback is sent to `ApplyOnStacksChange`?

Reply:
Yep it handles it all within the helper!